src/commands/run.js

```diff
@@ -29,9 +29,12 @@ class RunCommand extends ApifyCommand {
         const packageJsonPath = path.join(cwd, 'package.json');
         const mainPyPath = path.join(cwd, 'src/__main__.py');
 
+        const projectType = ProjectAnalyzer.getProjectType(cwd);
+        const actualStoragePath = process.env.APIFY_LOCAL_STORAGE_DIR || process.env.CRAWLEE_STORAGE_DIR || DEFAULT_LOCAL_STORAGE_DIR;
+
         const packageJsonExists = fs.existsSync(packageJsonPath);
         const mainPyExists = fs.existsSync(mainPyPath);
-        const isScrapyProject = ProjectAnalyzer.getProjectType(cwd) === PROJECT_TYPES.SCRAPY;
+        const isScrapyProject = projectType === PROJECT_TYPES.SCRAPY;
 
         if (!packageJsonExists && !mainPyExists && !isScrapyProject) {
             throw new Error(
```

Comment:
This kind of code looks like something I'd expect in `ProjectAnalyzer`. Couldn't the error be thrown if no project type matches? It could mean adding project types such as `PROJECT_TYPES.UNKNOWN_PYTHON` or `UNKNOWN_JAVASCRIPT`...

Reply:
That is outside the scope of this PR, but yes. That sort of logic _should've_ been in ProjectAnalyzer (alongside many things like handling purging of storages, etc). Ideally, we'd have a ticket for this to clean it up

Reply:
Make a ticket and you're off the hook then :slightly_smiling_face:

Reply:
Has been doneth: #476